engine/kv_engine.cpp

```diff
@@ -941,7 +941,18 @@ Status KVEngine::SDeleteImpl(Skiplist* skiplist, const StringView& user_key) {
               "Wrong existing record type while insert a delete reocrd for "
               "sorted collection");
           delayFree(OldDataRecord{ret.existing_record, new_ts});
-          delayFree(OldDeleteRecord{ret.write_record, new_ts});
+          if (ret.hash_entry_ptr != nullptr) {
```

Comment:
adding some comments for if-else case.

Reply:
here why not use `pointwithtag` for `ret.hash_entry_ptr`,  then check the tag.

Reply:
Will do this while optimize hash entry size to 8